### DISCUSSION TITLE
Log Swap events from DEX

### DISCUSSION THREAD

**Discussion by Industrial** [wevm/viem#1010] [Question] [answered]
Hello.

This is a well documented software but there's something that I'm missing. A complete example of how to get data on what is happening on a DEX like BaseSwap (Base chain).

What I want to do specifically is to create OHLCV (chart) data from events that happen on a contract like swaps from BaseSwap.

This is what I have:

```
const client = createPublicClient({
  chain: base,
  transport: http(),
})
client.watchEvent({
  address: '0x78a087d713Be963Bf307b18F2Ff8122EF9A63ae9',
  // onError: (error) => {
  //   console.error(error)
  // },
  onLogs: (logs) => {
    console.log(`${new Date().toISOString()} bswapContract`, logs)
  },
})
```

1. This logs nothing. It's been running for an hour and I get nothing logged.
2. When I enable the `onError` I get:
    ```
    Request body: {"method":"eth_getFilterChanges","params":["0xd0f3a6cf5dcb74d48ecc89cc22bc0d20"]}
    ...
    { code: -32000, message: "filter not found" }
    ```

Am I doing something wrong?

**Comment by ghost** [comment]
Did you provide the correct ABI event? What are you listening to, the Router or the single Pool / Pair?

**Reply by Industrial** [reply]
These are the contracts that I have. I am not sure which one I should be listening on, but I'll try the router :)

```
export const bswapContract = '0x78a087d713Be963Bf307b18F2Ff8122EF9A63ae9'
export const masterchefContract = '0x2B0A43DCcBD7d42c18F6A83F86D1a19fA58d541A'
export const factoryContract = '0xFDa619b6d20975be80A10332cD39b9a4b0FAa8BB'
export const routerContract = '0x327Df1E6de05895d2ab08513aaDD9313Fe505d86'
export const lockedLiquidityContract = '0x8fe53fa234ff89bed2ca6cb2c5528f53a6e5e3df313279d694421f4cef8cc4b2'
```

**Reply by ghost** [reply, marked as answer]
The `swap()` event is emitted by the pair contract. https://docs.uniswap.org/contracts/v2/reference/smart-contracts/pair
